src/pynxtools/nexus/nexus.py

```diff
@@ -425,6 +425,91 @@ def get_inherited_hdf_nodes(
     return (class_path, nxdl_elem_path, elist)
 
 
+def safe_str(value, precision: int = 8) -> str:
+    """Return a deterministic string representation of arrays, lists, or scalars.
+
+    Floats are formatted consistently across systems to ensure deterministic
+    output. Special handling is applied to simplify representation:
+      - `0.0` → `'0.0'`
+      - `1.0` → `'1'`
+      - `1.50` → `'1.5'`
+      - Non-integer floats keep up to `precision` decimals with trailing zeros
+        and dots removed.
+
+    Arrays and lists are formatted elementwise using the same rules.
+
+    Args:
+        value: The input value to format. Can be a scalar, list, tuple,
+            NumPy array, or basic type such as int, float, str, or bytes.
+        precision (int): Maximum number of decimal places for non-integer
+            floats. Defaults to 8.
+
+    Returns:
+        str: Deterministic string representation of the input.
+    """
+    # Normalize NumPy scalar and 0D array types
+    if isinstance(value, np.generic):
+        value = value.item()
+    elif isinstance(value, np.ndarray) and value.shape == ():
+        value = value.item()
+
+    def format_float(value: float) -> str:
+        """Format a float deterministically."""
+        if value == 0.0:
```

Comment:
Almost always epsilon comparison https://stackoverflow.com/questions/19837576/comparing-floating-point-number-to-zero

But what is the aim to achieve here, why not report all digits?
Indeed often floats hav a lower fractional parts that make two values different but that is a practical irrelevant difference.

Is this the rationale behind this formatting change?
Again then why not render all digits?